src/Provider/AbstractProvider.php

```diff
@@ -702,18 +702,26 @@ protected function parseResponse(ResponseInterface $response)
      * Custom mapping of expiration, etc should be done here. Always call the
      * parent method when overloading this method.
      *
-     * @param  mixed $result
-     * @return array
+     * @param  array|string $response
+     * @return array|string
+     * @throws UnexpectedValueException
      */
-    protected function prepareAccessTokenResponse(array $result)
+    protected function prepareAccessTokenResponse($response)
```

Comment:
I don't think so. The key thing to remember here is that request/response is more than just the OAuth flow, since we also have resource owners. In the case of `prepareAccessTokenResponse`, we absolutely know that the response must be an array. In other situations, requesting other API endpoints, that may not always be the case and we have a mixed result.

In this case, Scrutinizer simply doesn't have enough facts to make a judgement call about the code path in question. The warning should be ignored, because we can reason about the code path more completely than an automated test.